### ISSUE TITLE
rapAlignment option has no effect when extracting samples

### ISSUE THREAD

**Issue by richardmitic** [gpac/mp4box.js#30]

**Comment by cconcolato** [comment]
Thanks for the report. What would you expect if the file does not contain enough RAP ? or if the RAP are not aligned with the number of samples ? Should RAP=true have precedence over the number of samples requested ?

**Comment by richardmitic** [comment]
Yes, I think RAP=true should have precedence. That's how it works in the standard mp4box, so it seems sensible to align it. For example:
```bash
wget https://dl.dropboxusercontent.com/u/7245016/bug_buck_bunny_trailer.mp4
mp4box -dash 1 -rap bug_buck_bunny_trailer.mp4
```
will ignore the segment time and split at the RAPs.

I'm not quite sure what you mean by "not enough RAPs" but it would make sense for any remaining samples to be output when you reach the end of the file, so you could ensure that you can process all the samples in a file as long as you call `mp4box.flush()`.

**Comment by cconcolato** [comment]
Ok, that's what I thought too. Will implement it when I have time.

**Comment by IainCole** [comment]
+1

**Comment by collin** [comment]
I've been using `mp4box.js` with great success. But I've run into this issue as well.

I looked into the source for this and it looks like the `rapAlignement` option isn't checked at all while processing segments.

I was able to figure out a change that will honor that setting. When I went to work on test coverage I found the `.mp4` files for testing aren't in the repository so I'm not sure how to move forward.
